x-pack/plugins/security_solution/server/endpoint/services/cases/cases.ts

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { CommentType } from '@kbn/cases-plugin/common';
+import type { CasesByAlertId } from '@kbn/cases-plugin/common/api';
+import type { CasesClient } from '@kbn/cases-plugin/server';
+import { APP_ID } from '../../../../common';
+import type {
+  HostMetadata,
+  HostMetadataInterface,
+  ImmutableObject,
+} from '../../../../common/endpoint/types';
+import type { CreateActionPayload } from '../actions/create/types';
+
+export const updateCases = async ({
+  casesClient,
+  createActionPayload,
+  endpointData,
+}: {
+  casesClient?: CasesClient;
+  createActionPayload: CreateActionPayload;
+  endpointData: Array<ImmutableObject<HostMetadataInterface>>;
+}): Promise<void> => {
+  if (!casesClient) {
+    return;
+  }
+  // convert any alert IDs into cases
+  let caseIDs: string[] = createActionPayload.case_ids?.slice() || [];
+  if (createActionPayload.alert_ids && createActionPayload.alert_ids.length > 0) {
+    const newIDs: string[][] = await Promise.all(
+      createActionPayload.alert_ids.map(async (alertID: string) => {
+        const cases: CasesByAlertId = await casesClient.cases.getCasesByAlertID({
+          alertID,
+          options: { owner: APP_ID },
+        });
+        return cases.map((caseInfo): string => {
+          return caseInfo.id;
+        });
+      })
+    );
+    caseIDs = caseIDs.concat(...newIDs);
+  }
+  caseIDs = [...new Set(caseIDs)];
+
+  // Update all cases with a comment
+  if (caseIDs.length > 0) {
+    const targets = endpointData.map((endpoint: HostMetadata) => ({
+      hostname: endpoint.host.hostname,
+      endpointId: endpoint.agent.id,
+    }));
+
+    await Promise.all(
+      caseIDs.map((caseId) =>
+        casesClient.attachments.bulkCreate({
+          caseId,
+          attachments: [
+            {
+              type: CommentType.actions,
+              comment: createActionPayload.comment || '',
+              actions: {
+                targets,
+                type: createActionPayload.command,
+              },
+              owner: APP_ID,
+            },
+          ],
+        })
+      )
+    );
```

Comment:
Hey! There is no need to make one call per attachment. The `bulkCreate` will create the attachments in one call. You can do:

```suggestion
    const attachments = caseIDs.map((caseId) => ({
        type: CommentType.actions,
        comment: createActionPayload.comment || '',
        actions: {
          targets,
          type: createActionPayload.command,
        },
        owner: APP_ID,
      }));
      
      await casesClient.attachments.bulkCreate({caseId, attachments})
```

Reply:
Done e7fa45ad9f8183c7795f8727a60b6d5933248749